scripts/configure-venue-access.ts

```diff
@@ -0,0 +1,141 @@
+import fs from "fs";
+import admin from "firebase-admin";
+import serviceAccount from "./sparkle-7-firebase-adminsdk-f5a3m-482bebb057.json";
+import "firebase/firestore";
+import { Venue } from "../src/types/Venue";
+import {
+  VenueAccessCodes,
+  VenueAccessEmails,
+  VenueAccessType,
+} from "../src/types/VenueAcccess";
+
+const mergeStringArrays = (
+  arr1: string[] = [],
+  arr2: string[] = []
+): string[] => {
+  const result = (arr1 ?? []).map((val) => val.trim());
+  (arr2 ?? []).forEach((val) => {
+    const valTrimmed = val.trim();
+    if (!result.includes(valTrimmed)) {
+      result.push(valTrimmed);
+    }
+  });
+  return result;
+};
+
+function usage() {
+  console.log(`
+${process.argv[1]}: Configure venue access. Supports configuring a secret password, list of emails which can access the venue, or ticket codes.
+
+Usage: npx ts-node ${process.argv[1]} PROJECT_ID VENUE_ID [password|emaillist|codelist] [password | emails file path | codes file path]
+
+Example: npx ts-node ${process.argv[1]} co-reality-map password abc123
+Example: npx ts-node ${process.argv[1]} co-reality-map emails emails-one-per-line.txt
+Example: npx ts-node ${process.argv[1]} co-reality-map codes ticket-codes-one-per-line.txt
+`);
+  process.exit(1);
+}
+
+const argv = process.argv.slice(2);
+if (argv.length < 4) {
+  usage();
+}
+
+const [projectId, venueId, method, accessDetail] = argv;
```

Comment:
This doesn't follow our latest script patterns. See other scripts for examples on how this should be done.

Reply:
Can we add the patterns and conventions to the README.md in the scripts dir? It'd be much easier to follow them and build new scripts.

Reply:
Captured some of my notes on this on the documentation PR:

- https://github.com/sparkletown/sparkle/pull/1062#issuecomment-763287172

As mentioned on slack:

> I feel like it could get as outdated as quickly as it’s written, as this evolves fairly quickly/etc.
>
> My personal approach has been to look at the other scripts, particularly those that have been updated most recently, and to see the patterns/approach that is being taken there.